### ISSUE TITLE
Span/ReadonlySpan TryCopy/CopyTo are very slow

### ISSUE THREAD

**Issue by davidfowl** [dotnet/corefx#16840]
The implementation of TryCopyTo/CopyTo is extremely slow (byte by byte) for the portable span. There's a TODO to use a new API in Unsafe (which has since been added) but it was never implemented.

https://github.com/dotnet/corefx/blob/5ee9e2a7ded9d609e69bbd7c1ed6823b97737a72/src/System.Memory/src/System/Span.cs#L329

https://github.com/dotnet/corefx/blob/5ee9e2a7ded9d609e69bbd7c1ed6823b97737a72/src/System.Memory/src/System/ReadOnlySpan.cs#L226

/cc @jkotas @benaadams @ahsonkhan @KrzysztofCwalina @nietras

**Comment by KrzysztofCwalina** [comment]
@ahsonkhan is working on this.

**Comment by KrzysztofCwalina** [comment]
@ahsonkhan, while you are doing th eperf work. I would also like us to consider changing the signature of CopyTo. Whenever use CopyTo on Span, I wish the method Always succeeded, copied as much as it can, and returned an int indicating how much was copied.  

e.g. 
```c#
int copied = 0;
while(while copied < source.Length) {
    copied += source.Slice(copied).CopyTo(destination);
    destination = GetMoreSpace();
}
```
... and many similar scenarios. Without such CopyTo there is lots of ceremony around making sure the source is not larger than the destination.

@davidfowl, @terrajobst, @jkotas, any thoughts?

**Comment by jkotas** [comment]
We should not be hijacking this issue with orthogonal API design discussion. Opened(https://github.com/dotnet/corefx/issues/16888

**Comment by shiftylogic** [comment]
Fixed by #17063

**Comment by davidfowl** [comment]
@shiftylogic This needs to be implemented for `ReadOnlySpan<T>`

**Comment by ahsonkhan** [comment]
Fixed in #17297

**Comment by davidfowl** [comment]
👏
